freqtrade/data/dataprovider.py

```diff
@@ -117,18 +97,99 @@ def get_analyzed_dataframe(self, pair: str, timeframe: str) -> Tuple[DataFrame,
             combination.
             Returns empty dataframe and Epoch 0 (1970-01-01) if no dataframe was cached.
         """
-        if (pair, timeframe) in self.__cached_pairs:
-            return self.__cached_pairs[(pair, timeframe)]
+        pair_key = (pair, timeframe)
+        if pair_key in self.__cached_pairs:
+            if self.runmode in (RunMode.DRY_RUN, RunMode.LIVE):
+                df, date = self.__cached_pairs[pair_key]
+            elif pair_key in self.__sliced_pairs:
+                df, date = self.__sliced_pairs[pair_key]
+            else:
+                max_index = self.__slice_index
+                df, date = self.__cached_pairs[pair_key]
+                df = df.iloc[max(0, max_index - MAX_DATAFRAME_CANDLES):max_index]
+                self.__cached_pairs[pair_key] = df, date
```

Comment:
i think you'll want to assign this to `__sliced_pairs` - not `__cached_pairs` pairs ...

makes me wonder though - what's the point in caching the sliced version?
in dry/live - we wouldn't slice (no point - just return everything

in backtesting - you'll usually only query the same slice once (after that you're one candle further, so need a new slice) - so if you wanna cache the slice, the key needs to include the index (`max_index`?) too - but at that point - slicing will be rather pointless.